### PR TITLE
Adding the pycbc offline executable

### PR DESCRIPTION
**Pull request by tapaimarton**
This PR adds the pycbc_offline executable. In the current iteration, this generates the workflow up to the trigger merge jobs, both for data and injections for 3 detectors. The optimal_snr jobs are also working with the correct config files.

### CODE REVIEW

**Comment by ahnitz** [comment]
@tapaimarton Thanks Marton, I think this is a good starting point to incrementally add in working features for multi detector analysis. Once that's complete we can think about whether the old one is necessary or not, but it's possible they'll diverge in other ways.

**Comment by ahnitz** [comment]
We'll wait for @spxiwh to OK this as well.

File: bin/workflows/pycbc_offline [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+                                            veto_cat_files,
+                                            rdir['analysis_time/segment_data'],
+                                            execute_now=False)
+
+# setup gating files if provided
+gate_files = wf.setup_gating_workflow(workflow, output_dir="gating")
+
+# add the gate files to the jobs that use them
+for job in ['calculate_psd', 'inspiral', 'single_template',
+            'plot_singles_timefreq']:
```
Comment (spxiwh) [inline bin/workflows/pycbc_offline]: We need to delete this part and have it done automatically, but I'll merge that in later ...

File: bin/workflows/pycbc_offline [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+ind_insps = insps = wf.setup_matchedfltr_workflow(workflow, analyzable_segs,
+                                   datafind_files, splitbank_files_fd,
+                                   output_dir, tags = [tag])
+
+insps = wf.merge_single_detector_hdf_files(workflow, hdfbank[0],
+                                           insps, output_dir, tags=[tag])
+
+# setup sngl trigger distribution fitting jobs
+# 'statfiles' is list of files used in calculating coinc statistic
+statfiles = []
```
Comment (spxiwh) [inline bin/workflows/pycbc_offline]: I have a suspicion this is another function that can be removed, mostly writing this to remind me to look at this next week.

**Comment by duncan-brown** [comment]
@spxiwh the name of this executable is pretty obscure. Please can you rename it so it’s function is more obvious?

**Pull request merged**